### PR TITLE
Differential privacy training for private predictions

### PR DESCRIPTION
**Pull request by yanndupis**
Hey @mortendahl,

I have added a notebook showing how to train a model with differential privacy before serving private predictions. 

I have also replaced the simple dnn model in the existing example with a cnn model. 

Thanks,

### CODE REVIEW

File: examples/notebooks/keras-classification/a - Secure Classification with TFE Keras - Public Training - with DP.ipynb [hunk 1]
```diff
@@ -0,0 +1,291 @@
+{
```
Comment (mortendahl) [inline examples/notebooks/keras-classification/a - Secure Classification with TFE Keras - Public Training - with DP.ipynb]: <p>is this stuff relevant when running in a notebook?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tf-encrypted/tf-encrypted/pull/642/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: examples/notebooks/keras-classification/b - Secure Classification with TFE Keras - Secure Model Serving .ipynb [hunk 2]
```diff
@@ -16,7 +16,9 @@
      "name": "stderr",
```
Comment (mortendahl) [inline examples/notebooks/keras-classification/b - Secure Classification with TFE Keras - Secure Model Serving .ipynb]: <p>imports could be cleaned up</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tf-encrypted/tf-encrypted/pull/642/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (yanndupis) [reply examples/notebooks/keras-classification/b - Secure Classification with TFE Keras - Secure Model Serving .ipynb]: <p>ah good catch!</p><p> </p>

File: examples/notebooks/keras-classification/a - Secure Classification with TFE Keras - Public Training - with DP.ipynb (hunk 1 above)
Comment (yanndupis) [reply examples/notebooks/keras-classification/a - Secure Classification with TFE Keras - Public Training - with DP.ipynb]: <p>good point, I can remove the flag approach</p>

**Comment by yanndupis** [comment]
Thank you @mortendahl for your review. All the comments should be addressed now.

**Pull request merged**